phoenix-core/src/main/java/org/apache/phoenix/expression/JsonPathAsElementExpression.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.expression;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.expression.visitor.ExpressionVisitor;
+import org.apache.phoenix.schema.json.PhoenixJson;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PJson;
+import org.apache.phoenix.schema.types.PVarchar;
+
+
+public class JsonPathAsElementExpression extends BaseJSONExpression{
+	private PDataType datatype=null;
+	public JsonPathAsElementExpression(List<Expression> children) {
+        super(children);
+    }
+	public JsonPathAsElementExpression() {
+    }
+	@Override
+	public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr)  {
+		if (!children.get(1).evaluate(tuple, ptr)) {
+            return false;
+        }
+		String[] pattern =decodePath((String)PVarchar.INSTANCE.toObject(ptr));
+		if (!children.get(0).evaluate(tuple, ptr)) {
+	        return false;
+	    }
+		PhoenixJson value = (PhoenixJson) PJson.INSTANCE.toObject(ptr);
```

Comment:
Can you change the rhs to be array. This will be consistent with how you implemented ?| and ?&